### DISCUSSION TITLE
Journaling (ChatGPT plugin)

### DISCUSSION THREAD

**Discussion by ghost** [brianpetro/obsidian-smart-connections#111] [Ideas]
It would be awesome if the plugin could combine OpenAI ChatGPT API & Pinecone. This could allow users to interact with OpenAI chatgpt and use it as a daily journaling tool. Pinecone could be used to handle the memory to ensure ChatGPT can remember any previous dialogues. All dialogues could constantly be saved in the "today's daily note".

Think this would be very valuable for a lot of people :)

**Comment by brianpetro** [comment]
Hi @jacob1983 , and welcome to the Smart Connections discussion!

For a little background, Smart Connections handles what Pinecone does, except on your computer instead of the cloud. I chose to do it this way to maintain parity with Obsidian's mission of making your notes local first.

However, it sounds like you're describing a ChatGPT plugin that integrates with your Obsidian. And this is something I've been considering developing.

Is anyone else interested in something like this? If so, please leave a comment!

**Reply by ghost** [reply]
> Hi @jacob1983 , and welcome to the Smart Connections discussion!
> 
> For a little background, Smart COnnections handles what Pinecone does, except on your computer instead of the cloud. I chose to do it this way to maintain parity with Obsidian's mission of making your notes local first.
> 
> However, it sounds like you're describing a ChatGPT plugin that integrates with your Obsidian. And this is something I've been considering developing.
> 
> Is anyone else interested in something like this? If so, please leave a comment!

Hi! 

Thanks for your swift reply Brian :)

You are right, what I am suggesting is a way to use GPT as a speaking partner of a sort and then automatically save those conversations as a record in Obsidian. My own use case for this would be to use GPT as a form of coach for daily journaling (similar to what is described here https://every.to/chain-of-thought/gpt-3-is-the-best-journal-you-ve-ever-used) but there are for sure many other use cases since GPT is used for everything from simulated job interviews to the creation of food recipes.

A memory based on previous conversations (either locally or in the cloud) would certainly be of help in use cases where a long-time ongoing dialogue is wished for.

**Comment by abdcef** [comment]
I thought there are a few plugins that do that already, no? 
It is just too much work to keep fine-tuning the prompts to make them do the right thing. I got tired after the initial euphoria wore off..

**Reply by brianpetro** [reply]
My understanding is that @jacob1983 is interested in using the ChatGPT interface provided by OpenAI.

Smart Connections does already integrate with the "ChatGPT API" provided by OpenAI.
